scipy/ndimage/tests/test_interpolation.py

```diff
@@ -54,128 +69,156 @@ def shift(x):
+        xp_assert_equal(
             ndimage.geometric_transform(data, shift, cval=-1, mode=mode,
-                                        output_shape=(4,)))
+                                        output_shape=(4,)),
+            xp.asarray(expected_value))
 
     @pytest.mark.parametrize('mode', ['mirror', 'reflect', 'grid-mirror',
                                       'grid-wrap', 'grid-constant',
                                       'nearest'])
     @pytest.mark.parametrize('order', range(6))
-    def test_boundary_spline_accuracy(self, mode, order):
+    def test_boundary_spline_accuracy(self, mode, order, xp):
         """Tests based on examples from gh-2640"""
-        data = np.arange(-6, 7, dtype=float)
-        x = np.linspace(-8, 15, num=1000)
-        y = ndimage.map_coordinates(data, [x], order=order, mode=mode)
+        data = xp.arange(-6, 7, dtype=xp.float64)
+        x = xp.linspace(-8, 15, num=1000)
+        y = ndimage.map_coordinates(data, xp.asarray([x]), order=order, mode=mode)
 
         # compute expected value using explicit padding via np.pad
         npad = 32
         pad_mode = ndimage_to_numpy_mode.get(mode)
-        padded = np.pad(data, npad, mode=pad_mode)
-        expected = ndimage.map_coordinates(padded, [npad + x], order=order,
+        padded = xp.asarray(np.pad(np.asarray(data), npad, mode=pad_mode))
+        expected = ndimage.map_coordinates(padded, xp.asarray([npad + x]), order=order,
                                            mode=mode)
 
         atol = 1e-5 if mode == 'grid-constant' else 1e-12
-        assert_allclose(y, expected, rtol=1e-7, atol=atol)
+        xp_assert_close(y, expected, rtol=1e-7, atol=atol)
 
     @pytest.mark.parametrize('order', range(2, 6))
     @pytest.mark.parametrize('dtype', types)
-    def test_spline01(self, dtype, order):
-        data = np.ones([], dtype)
+    def test_spline01(self, dtype, order, xp):
+        dtype = getattr(xp, dtype)
+        data = xp.ones([], dtype=dtype)
         out = ndimage.spline_filter(data, order=order)
-        assert_array_almost_equal(out, 1)
+        assert out == xp.asarray(1, dtype=out.dtype)
 
     @pytest.mark.parametrize('order', range(2, 6))
     @pytest.mark.parametrize('dtype', types)
-    def test_spline02(self, dtype, order):
-        data = np.array([1], dtype)
+    def test_spline02(self, dtype, order, xp):
+        dtype = getattr(xp, dtype)
+        data = xp.asarray([1], dtype=dtype)
         out = ndimage.spline_filter(data, order=order)
-        assert_array_almost_equal(out, [1])
+        assert_array_almost_equal(out, xp.asarray([1]))
 
     @pytest.mark.parametrize('order', range(2, 6))
     @pytest.mark.parametrize('dtype', types)
-    def test_spline03(self, dtype, order):
-        data = np.ones([], dtype)
+    def test_spline03(self, dtype, order, xp):
+        if xp.__name__ == 'array_api_strict':
+            pytest.xfail("array_api_strict dtypes cannot be used for output=")
```

Comment:
could you explain what the problem is here?

Reply:
TL:DR: This properly belongs to https://github.com/scipy/scipy/pull/21150

To make long story short: `ndimage` functions  accept the `output` argument which can be either an array to place the result _or_ a dtype to allocate the output array. The current version of https://github.com/scipy/scipy/pull/21150 keeps internal ndimage functions numpy-specific (because they are mostly compiled code); and attempting to feed an `array_api_strict` dtype in place of a numpy dtype does not go well.

Reply:
Note that `output=` should remain numpy-only, it's not supportable in general (and rarely used / not a great usage pattern). Similar to how `out=` for numpy functions isn't part of the array API standard.